src/main/java/com/icthh/xm/ms/configuration/repository/impl/ConfigProxyRepository.java

```diff
@@ -245,8 +243,25 @@ public void refreshTenant(String tenant, ConfigurationList configs) {
     }
 
     @Override
-    public void refreshTenant(String tenant) {
-        refreshTenant(tenant, null);
+    public void refreshTenants(List<String> tenants, ConfigurationList configs) {
+        final ConfigurationList configurationList = configs == null ? persistenceConfigRepository.findAll() : configs;
+        Map<String, Set<String>> updates = new HashMap<>();
+        tenants.stream().forEach((tenant) -> {
+            long start = System.currentTimeMillis();
+            List<Configuration> actualConfigs = configurationList.getData().stream()
+                .filter(config -> config.getPath().startsWith(getTenantPathPrefix(tenant)))
+                .collect(toList());
+
+            Set<String> oldKeys = storage.getConfigPathsList(tenant);
+            Set<String> updated = storage.refreshStorage(actualConfigs, oldKeys);
+            updates.put(tenant, updated);
+        });
+        storage.reprocess(tenants.get(0));
```

Comment:
why we reprocess only first tenant?

Reply:
Because actually it reprocesses the parent, and parent is the same for all tenants from tenats list here. So, we do not need to reprocess them all.